### PR TITLE
Bug 1370594: Move and style newsletter for redesign

### PR DESCRIPTION
**Pull request by stephaniehobson**
Things to test:
- [x] newsletter takes up full width of page immediately above footer
- [x] at 1200px the newsletter box becomes a 2 column layout
- [x] privacy checkbox (revealed when input gets focus) isn't broken
- [x] error message (revealed if you use the email address failure@example.com) isn't broken
- [x] success message (reveled if you use the email success@example.com) isn't broken 
- [x] other stuff you'd normally test to make sure I'm paying attention ;)
- [x] looks good on homepage still
- [x] look good on user profile edit page still

Notes:
- you will have to delete your local storage variable newsletterHide to see it again after submitting

Spec:
- form field and button adjustments are coming in later PR
- increased spacing is also coming in later PR
screen shot 2017-06-13 at 15 04 35

### CODE REVIEW

**Comment by schalkneethling** [comment]
I submitted the form using the fail address but got the following:

screen shot 2017-06-14 at 15 30 09

**Comment by schalkneethling** [comment]
Also getting a success with the fail@example.com address using the old form but, here the form is replaced by the messaging instead of appending it below the form.

**Comment by schalkneethling** [comment]
Oh ok, so this is overriding `.hidden`

```
@media (min-width: 1200px)
.newsletter-box .newsletter-form {
    display: grid;
    grid-column-gap: 20px;
    grid-template-columns: 2fr 1fr;
    grid-template-rows: auto auto;
}
```

File: kuma/wiki/jinja2/wiki/document.html [hunk 1]
```diff
@@ -383,6 +383,12 @@ <h2 class="offscreen">{{ _('Document Tags and Contributors') }}</h2>
       </div>
     </div> <!-- ends "main-content" -->
 
+    {% if waffle.switch('newsletter') and waffle.switch('newsletter_article') and waffle.flag('redesign') %}
```
Comment (schalkneethling) [inline kuma/wiki/jinja2/wiki/document.html]: Does it matter that I do not have the `newsletter` and `newsletter_article` flags? I do have `redesign` enabled however.

**Comment by schalkneethling** [comment]
On the front page the display looks fine however, `fail@example.com` still passes through:

screen shot 2017-06-14 at 15 39 58 1

**Comment by schalkneethling** [comment]
Hmmm, I do not see the newsletter on this page for example:
http://localhost:8000/en-US/profiles/sneethling@mozilla.com

This is a profile page so it should show right?

**Comment by schalkneethling** [comment]
> at 1200px the newsletter box becomes a 2 column layout

You mean two rows? i.e. stacked?

**Comment by schalkneethling** [comment]
In IE10 it defaults to the stacked layout irrespective of the viewport width correct?

Also, got this weird thing in IE10(perhaps a VM thing) where after I clicked submit, I got sent to https://www.mozilla.org/en-US/newsletter/

**Comment by stephaniehobson** [comment]
Hmmm. Looks like this PR is missing one of my final commits :( I'll try pushing again.

**Comment by stephaniehobson** [comment]
Nope, that wasn't it. I know I fixed the sucess message showing up with the form though. Need to dig around.

File: kuma/wiki/jinja2/wiki/document.html (hunk 1 above)
Comment (stephaniehobson) [inline kuma/wiki/jinja2/wiki/document.html]: These are switches not flags (different part of admin console) but, you should not be seeing the newsletter if you don't have these added and enabled. Maybe they came with your database dump?

Comment (schalkneethling) [inline kuma/wiki/jinja2/wiki/document.html]: Gotcha! must be yes, as I do see the newsletter.

File: kuma/static/styles/components/newsletter.scss [hunk 2]
```diff
@@ -39,16 +29,94 @@
...
+
+            .newsletter-form {
+                display: grid;
+                grid-column-gap: $grid-spacing;
+                grid-template-columns: 2fr 1fr;
+                grid-template-rows: auto auto;
+            }
+
+            &.hidden {
+                display: none;
```
Comment (stephaniehobson) [inline kuma/static/styles/components/newsletter.scss]: Do you see this line in your local copy? It should have fixed the message appearing conflict.

**Comment by stephaniehobson** [comment]
@schalkneethling I got the testing email wrong, it's failure@example.com >_< Also see my question in the comment above where I should have @ed you ^^

**Comment by stephaniehobson** [comment]
Also, it's the profile *edit* page, not the profile page. My bad.

File: kuma/static/styles/components/newsletter.scss (hunk 2 above)
Comment (schalkneethling) [inline kuma/static/styles/components/newsletter.scss]: I do see it now, not sure if you pushed another commit. Will test and see if I still get the same behaviour.

Comment (schalkneethling) [inline kuma/static/styles/components/newsletter.scss]: Ah ok, I see the problem now. You need to move the `&.hidden` into the `.newsletter-form` block:

```
.newsletter-form {
                display: grid;
                grid-column-gap: $grid-spacing;
                grid-template-columns: 2fr 1fr;
                grid-template-rows: auto auto;

                &.hidden {
                    display: none;
                }
            }
```

**Comment by schalkneethling** [comment]
Other than my comment here https://github.com/mozilla/kuma/pull/4263/files#r122159951 ~ This is looking good~ r+w/c 🗞

File: kuma/wiki/jinja2/wiki/document.html (hunk 1 above)
Comment (stephaniehobson) [inline kuma/wiki/jinja2/wiki/document.html]: I think this is fixed now. Test again?

**Pull request merged**